scipy/interpolate/_bsplines.py

```diff
@@ -561,6 +561,87 @@ def integrate(self, a, b, extrapolate=None):
         integral *= sign
         return integral.reshape(ca.shape[1:])
 
+    def design_matrix(self, x, kind="dense", check_finite=True):
```

Comment:
And let's just always return a csr matrix (keep the dense equivalence in the tests and in a docstring example)